python/ql/test/experimental/dataflow/tainttracking/unwanted-global-flow/test.py

```diff
@@ -0,0 +1,32 @@
+import sys; import os; sys.path.append(os.path.dirname(os.path.dirname((__file__))))
+from taintlib import *
+
+# Various instances where flow is undesirable
+
+tainted = NOT_TAINTED
```

Comment:
\*confused noises\*

I see they are just using the same naming as before, but oh man, this one made my scratch my head.

Can we maybe just call them `x` and `y` instead of `tainted` and `tainted2`? (or `ends_up_being_tainted_at_some_point`, and `ends_up_being_tainted_at_some_point2` :smile:)

_Then I promise I won't annoy you with more changes to these test files :sweat_smile:_

![scract my head](https://media.giphy.com/media/26tP4gFBQewkLnMv6/source.gif)

Reply:
I have rewritten it to hopefully make it more clear what the tests are testing and why. Also changed the second test case a bit so that it actually highlights the fundamental problem with global flow.